miden-lib/asm/miden/kernels/tx/faucet.masm

```diff
@@ -60,7 +63,8 @@ export.mint_fungible_asset
     # => [amount, TOTAL_ISSUANCE, ASSET]
 
     # update the total issuance
-    add exec.account::get_faucet_storage_data_slot exec.account::set_item dropw
+    # note: storage slot is known here, so set_item_raw can be used
+    add exec.account::get_faucet_storage_data_slot exec.account::set_item_raw dropw
```

Comment:
I am not sure this is going to work. If I'm understanding the flow correctly:

- `account::set_item_raw` will not fire `STORAGE_ON_AFTER_SET_ITEM` event.
- This means that `Host::on_account_storage_after_set_item()` won't get executed and the `AccountDeltaTracker` won't be updated.
- This means that changes to the account storage for the faucet reserved slot won't be included in account delta.

We should probably modify one of the tests to test for this (i.e., when we mint a new asset, account storage delta should include the update incrementing the total number of assets minted).

Reply:
One potential way to handle this is to introduce new events which get fired in `mint_asset` and `burn_asset` procedures in `api.masm`. Then handlers of these events would modify the account delta tracker accordingly.

Though, this may be introducing too much complexity.

Reply:
I was thinking about the same. If this change ("firing all events from `api.masm`") is purely aesthetics, we should not introduce this extra complexity. If the change adds readability to the transaction kernel, we should make the change.

One problem is that in `api.masm`, we don't have the information that we want to update yet. To update the `account_storage_delta` in the correct way, we need the `TOTAL_ISSUANCE`, which we compute at a later stage. So, to fire all events in `api.masm`, we need to rewrite the way how the faucet works.